mico-core/src/main/java/io/github/ust/mico/core/service/MicoKubernetesClient.java

```diff
@@ -263,49 +268,189 @@ public Service createMicoServiceInterface(MicoServiceInterface micoServiceInterf
+            log.debug("There are no service deployment information with Kubernetes deployment information " +
+                    "for the MicoApplication '{}' '{}' stored in the database. MicoApplication is considered to be not deployed.",
                 micoApplication.getShortName(), micoApplication.getVersion());
-        }
-
-        String deploymentStatus = result ? "deployed" : "not deployed";
-        log.info("MicoApplication '{}' in version '{}' is {}.",
-            micoApplication.getShortName(), micoApplication.getVersion(), deploymentStatus);
-        return result;
+   			return MicoApplicationDeploymentStatus.undeployed("MicoApplication is currently not deployed.");
+   		}
+   		
+    	// Default application deployment status is 'deployed'. The following code will
+   		// use a strategy which checks whether there are errors or inconsistencies within
+   		// the stored deployment information and the actual Kubernetes deployments and services.
+   		// If there are such errors or inconsistencies, this variable will be updated.
+   		MicoApplicationDeploymentStatus.Value applicationDeploymentStatus = Value.DEPLOYED;
+   		List<MicoMessage> messages = new ArrayList<>();
+   		String msg;
+   		
+   		// Check whether there are jobs currently running for the deployment of the MicoApplication
+   		MicoApplicationJobStatus applicationJobStatus = backgroundJobBroker
+		    .getJobStatusByApplicationShortNameAndVersion(micoApplication.getShortName(), micoApplication.getVersion());
+   		switch (applicationJobStatus.getStatus()) {
+   		case PENDING:
+   			// 'Pending' indicates that the deployment is scheduled to be executed in the future
+   			applicationDeploymentStatus = Value.PENDING;
+   			msg = "The deployment of MicoApplication '" + micoApplication.getShortName()
+                + "' '" + micoApplication.getVersion() + "' is scheduled to be started.";
+   			log.debug(msg);
+   			messages.add(MicoMessage.error(msg));
+   			return new MicoApplicationDeploymentStatus(applicationDeploymentStatus, messages);
+   		case RUNNING:
+   			// 'Running' indicates that the deployment is currently in progress
+   			applicationDeploymentStatus = Value.PENDING;
+   			msg = "The deployment of MicoApplication '" + micoApplication.getShortName()
+                + "' '" + micoApplication.getVersion() + "' is currently in progress.";
+   			log.debug(msg);
+   			messages.add(MicoMessage.error(msg));
+   			return new MicoApplicationDeploymentStatus(applicationDeploymentStatus, messages);
+   		case ERROR:
+   			// 'Error' indicates that errors occurred during the deployment
+            applicationDeploymentStatus = Value.INCOMPLETED;
+            msg = "The deployment of MicoApplication '" + micoApplication.getShortName()
+                + "' '" + micoApplication.getVersion() + "' failed.";
+            log.debug(msg);
+            messages.add(MicoMessage.error(msg));
```

Comment:
Consider to add the error messages of the background jobs to the error message of the deployment status (`applicationJobStatus.getJobs().get(index).getErrorMessage() != null`)